main.go

```diff
@@ -219,3 +203,44 @@ func encodeConfig(cfg *configv1alpha1.Configuration) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+func apply(configFile string) (ctrl.Options, configv1alpha1.Configuration) {
+	var err error
+	options := ctrl.Options{
+		Scheme: scheme,
```

Comment:
If we set values here, it will not be overwrite by ComponentConfig. So I moved to setOptionDefaults.
https://github.com/kubernetes-sigs/controller-runtime/blob/15154aaa67679df320008ed45534f83ff3d6922d/pkg/manager/manager.go#L456-L482